mne/time_frequency/tests/test_spectrum.py

```diff
@@ -295,44 +432,135 @@ def _check_spectrum_equivalent(spect1, spect2, tmp_path):
+            Klass(data, info, freqs, events)
+    # test unspecified method for unaggregated spectra (i.e. with segments or tapers)
+    if (
+        method == "welch"
+        and not average
+        or method == "multitaper"
+        and output == "complex"
+    ):
+        with pytest.raises(
+            ValueError, match="Invalid value for the 'method' parameter"
+        ):
+            Klass(data, info, freqs, method="unknown", mt_weights=mt_weights)
+    # test unspecified/mismatched multitaper weights
+    if method == "multitaper" and output == "complex":
+        with pytest.raises(
+            ValueError, match=r"Expected size of `mt_weights` to be.*, got"
+        ):
+            Klass(data, info, freqs, method=method, mt_weights=None)
+        with pytest.raises(
+            ValueError, match=r"Expected size of `mt_weights` to be.*, got"
+        ):
+            Klass(data, info, freqs, method=method, mt_weights=mt_weights[:, :-1])
 
 
 @pytest.mark.parametrize("kind", ("raw", "epochs"))
-def test_spectrum_array(kind, tmp_path, request):
+@pytest.mark.parametrize(
+    "method, output, average",
+    [
+        ("welch", "power", "mean"),  # test with precomputed spectrum
+        ("welch", "power", False),
+        ("welch", "complex", False),
+        ("welch", "complex", "mean"),
+        ("multitaper", "complex", None),
+    ],
+)
+def test_spectrum_array(kind, method, output, average, tmp_path, request):
     """Test EpochsSpectrumArray and SpectrumArray constructors."""
-    spectrum = request.getfixturevalue(f"{kind}_spectrum")
+    if method == "welch" and output == "power" and average:
+        spectrum = request.getfixturevalue(f"{kind}_spectrum")
+    else:
+        data = request.getfixturevalue(kind)
+        kwargs = dict()
+        if method == "welch":
+            kwargs.update(average=average)
+        spectrum = data.compute_psd(method=method, output=output, **kwargs)
     data, freqs = spectrum.get_data(return_freqs=True)
     Klass = SpectrumArray if kind == "raw" else EpochsSpectrumArray
-    spect_arr = Klass(data=data, info=spectrum.info, freqs=freqs)
+    spect_arr = Klass(
+        data=data,
+        info=spectrum.info,
+        freqs=freqs,
+        method=method,
+        mt_weights=spectrum.mt_weights,
+    )
     _check_spectrum_equivalent(spectrum, spect_arr, tmp_path)
 
 
 @pytest.mark.parametrize("kind", ("raw", "epochs"))
 @pytest.mark.parametrize("array", (False, True))
-def test_plot_spectrum(kind, array, request):
+@pytest.mark.parametrize(
+    "method, output, average",
+    [
+        ("welch", "power", "mean"),  # test with precomputed spectrum
+        ("welch", "power", False),
+        ("welch", "complex", False),
+        ("welch", "complex", "mean"),
+        ("multitaper", "complex", None),
+    ],
+)
+def test_plot_spectrum(kind, array, method, output, average, request):
     """Test plotting (Epochs)Spectrum(Array)."""
```

Comment:
Similar situation here for the plotting as above in terms of the various types of data that need to be checked, i.e. for aggregating over tapers/segments, converting complex -> power (warning that imag parts are being discarded when plotting would be raised otherwise).

Reply:
Can drop this down to only 3 runs whilst maintaining coverage.